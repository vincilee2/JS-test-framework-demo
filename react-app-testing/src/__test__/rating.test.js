import renderer from 'react-test-renderer';
import Rating from '../rating.js';

it('render Rating correctly with default color', () => {
    const tree = renderer.create(<Rating rating={5}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('render Rating correctly with specific color', () => {
    const tree = renderer.create(<Rating rating={5} color={'#EF0000'}/>).toJSON();
    expect(tree).toMatchSnapshot();
});