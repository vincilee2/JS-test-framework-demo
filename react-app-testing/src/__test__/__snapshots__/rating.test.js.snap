// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Rating correctly with default color 1`] = `
<div
  className="star-rating"
>
  <div
    className="star-rating-top"
    style={
      {
        "color": "#EE0000",
        "width": "50%",
      }
    }
  >
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
  <div
    className="star-rating-bottom"
  >
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
</div>
`;

exports[`render Rating correctly with specific color 1`] = `
<div
  className="star-rating"
>
  <div
    className="star-rating-top"
    style={
      {
        "color": "#EF0000",
        "width": "50%",
      }
    }
  >
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
  <div
    className="star-rating-bottom"
  >
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
</div>
`;
